### ISSUE TITLE
Pedantic Solar Thermal

### ISSUE THREAD

**Issue by thorade** [UdK-VPT/BuildingSystems#22]
```
Advanced.PedanticModelica:=true;
checkModel("BuildingSystems.Technologies.SolarThermal");
```

**Comment by carlesRT** [comment]
There were two main problems.

1. Output connector using the type prefix parameter. The error obtained was:

```Modelica
In model collector the component (Input or output connector) had attribute parameter .
That is not correct Modelica.
```

It is solved with a3e5a0b and 5d00a56. the type prefix parameter was deleted. Notice that these output connectors are still considered parameters in the models because they have the following annotation `Dialog(group="`Geometry"). 

Alternative one could remove the above annotation, add a parameter and link output and parameter (Analog to `Modelica.Blocks.Sources.Constant`).

2. Wrong class definition. The error obtained was:

```Modelica
Redeclaration requires a subtype.
But different specialized classes model collectorData and record collectorData.
```

Solved in 5823500.

**Comment by carlesRT** [comment]
merged 047dce9 - issue can be closed
